packages/button/src/Button.tsx

```diff
@@ -70,65 +53,61 @@ const Button: ExtendableBox<ButtonProps & { ref?: React.Ref<any> }, 'button'> =
+        return 'a';
+      } else {
+        return isAnchor ? 'a' : 'button';
+      }
+    }, [asProp, isAnchor]);
+
+    const rootProps = {
+      ...rest,
+      href: disabled ? '' : (rest.href as string | UrlObject),
+      onClick: !disabled ? onClick : undefined,
+      // only add the disabled prop if this is a button
+      ...(typeof rest.href !== 'string' && { disabled }),
+    } as const;
 
-    const buttonProps = {
+    const boxProps = {
       type: isAnchor ? undefined : type || 'button',
       className: cx(buttonClassName, className),
       ref: forwardRef,
-      // Provide a default value for the as prop
-      // If consuming application passes a value for as, it will override the default set here
-      as: as
-        ? as
-        : ((isAnchor ? 'a' : 'button') as keyof JSX.IntrinsicElements),
-      // only add a disabled prop if not an anchor
-      ...(typeof rest.href !== 'string' && { disabled }),
       'aria-disabled': disabled,
-      onClick: !disabled ? onClick : undefined,
+      as: finalAsProp,
       ...ButtonDataProp.prop,
-      ...rest,
+      // If this is not a Component, then the Box is also the root
+      ...(!isAsComponent(asProp) && rootProps),
     } as const;
 
-    const iconProps = { variant, size, darkMode, disabled, isIconOnlyButton };
-
-    const content = (
-      <>
-        {/* Ripple cannot wrap children, otherwise components that rely on children to render dropdowns will not be rendered due to the overflow:hidden rule. */}
-        <div className={cx(rippleStyle)} ref={rippleRef} />
-
-        <div
-          className={cx(buttonContentStyle, buttonContentSizeStyle[size], {
-            [css`
-              justify-content: space-between;
-            `]: !!rightGlyph && darkMode,
-          })}
-        >
-          {leftGlyph && (
-            <ButtonIcon
-              glyph={leftGlyph}
-              className={css`
-                justify-self: right;
-              `}
-              {...iconProps}
-            />
-          )}
-
-          {children}
+    const contentProps = {
+      rightGlyph,
+      leftGlyph,
+      darkMode,
+      disabled,
+      variant,
+      size,
+    } as const;
 
-          {rightGlyph && (
-            <ButtonIcon
-              glyph={rightGlyph}
-              className={css`
-                justify-self: left;
-              `}
-              {...iconProps}
-            />
-          )}
-        </div>
-      </>
+    const WrapperComponent = ({ ...props }) =>
+      isAsComponent(asProp) ? (
+        React.createElement(asProp, { href: props.href, ...props })
+      ) : (
+        <>{props.children}</>
+      );
+
+    return (
+      // If WrapperComponent is a Fragment then is ignored, and rootProps gets applied to the Box
+      <WrapperComponent {...rootProps}>
+        <Box {...boxProps}>
+          <ButtonContent {...contentProps}>{children}</ButtonContent>
+        </Box>
+      </WrapperComponent>
```

Comment:
Is there a reason why you're creating a `WrapperComponent` here instead of just passing the `asProp` to the `Box`?

Reply:
NextJS Link needs an anchor inside, so we need essentially two wrapper components.
`<Button as={Link} />` returns `<Link href><a>{children}</a></Link>`

Reply:
I actually resolved two other issues involving NextJS Link components as the `as` prop, and this is the [workaround](https://jira.mongodb.org/browse/PD-1966?focusedCommentId=4717266&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4717266) that I found.

If this works on your end, I'm wondering if a more graceful solution would either be to 1) add this to our guidelines overall and push engineers to this approach if they want to use NextJS Links, or 2) bake this behavior directly into the `<Box>` so all instances of NextLinks are replaced with the workaround approach.

Reply:
It's an issue that we see across many components, so finding one resolution would be nice.

Reply:
If I understand correctly, I don't think we'd want to force external developers to create a `Linker` component when the ideal solution would be that LG components could accept the NextJS `Link` component. 
I do think this is functionality that we could add more globally though (i.e. to `Box`)

Reply:
I took a swing at it, but I feel like the TS wizardry required to incorporate this into the `Box` component is out of scope of this ticket.